Amazon.IonDotnet/Internals/Binary/SystemBinaryReader.cs

```diff
@@ -314,16 +314,15 @@ public override IEnumerable<SymbolToken> GetTypeAnnotationSymbols()
 
         public override bool HasAnnotation(string annotation)
         {
-            if (annotation == null)
-            {
-                throw new ArgumentNullException(nameof(annotation));
-            }
-
             int? annotationId = null;
             foreach (int aid in Annotations)
             {
                 string text = GetSymbolTable().FindKnownSymbol(aid);
-                if (text == null)
+                if (annotation == null && text == null && aid == 0)
+                {
+                    return true;
+                }
+                else if (text == null)
```

Comment:
Nit: the text == null check exists twice here, so this can be reordered to be a bit better.

Reply:
Addressed